src/app/lists/EditModal.tsx

```diff
@@ -55,25 +57,70 @@ const listQuantityNoteAndLabels: [keyof Schema["lists"], keyof Schema["lists"],
 const EditModal: React.FC<Props> = ({ data, onClose }) => {
     const [toSubmit, setToSubmit] = useState<Partial<Schema["lists"]>>(data ?? {});
 
-    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
     const setKey = (event: React.ChangeEvent<HTMLInputElement>, key: string): void => {
         const newValue = event.target.value;
         setToSubmit({ ...toSubmit, [key]: newValue });
     };
 
     const onSubmit = async (): Promise<void> => {
-        const table = supabase.from("lists");
-        const { error } =
-            data === null
-                ? await table.insert(toSubmit)
-                : await table.update(toSubmit).eq("primary_key", toSubmit.primary_key);
-
-        if (error) {
-            setErrorMsg(error.message);
+        if (data === null) {
+            const { data: returnedListData, error: insertListItemError } = await supabase
```

Comment:
Should we pull the insert and update into two separate methods here so that we have something like
```ts
if (data === null) {
    addListItem(listItem)
} else {
    updateListItem(listItem)
}
```